cmd/esc/cli/env_set.go

```diff
@@ -71,6 +73,14 @@ func newEnvSetCmd(env *envCommand) *cobra.Command {
 				return fmt.Errorf("value looks like a secret; rerun with --secret to mark it as such, or --plaintext if you meant to leave it as plaintext")
 			}
 			if secret {
+				if yamlValue.Kind == yaml.ScalarNode && yamlValue.Tag != "!!str" {
+					err = yaml.Unmarshal([]byte(strconv.Quote(args[1])), &yamlValue)
+					if err != nil {
+						return errors.New("secret values must be strings; try surrounding the argument in both single and double quotes (e.g. '\"foo\"')")
```

Comment:
Will this change anything? Given that presumably, you have just done this internally (quoted the string). Appreciate the point of the error might be just to get someone to re-evaluate the argument they have passed, which if so fair enough. That said, if we're doing that -- perhaps include the value they passed in the error message?

Reply:
See my response to @lblackstone below--this is a bit of paranoia on my part to handle the case where auto-quoting doesn't work properly.